src/Parsing/Test/FStringTests.cs

```diff
@@ -0,0 +1,368 @@
+                        break;
+                }
+            }));
+            found.Should().BeTrue();
+        }
+
+        [TestMethod, Priority(0)]
+        public void FormattedStringPlus() {
+            var code = "'a mess with 'f'concat of {some}' 'strings' 'and {things}'";
+            var parser = Parser.CreateParser(MakeReader(code), PythonLanguageVersion.V36);
+            var ast = parser.ParseFile();
+            var foundSome = false;
+            var foundThings = false;
+            ast.Walk(new MyPythonWalker(expr => {
+                switch (expr) {
+                    case NameExpression nameExpr:
+                        foundSome |= nameExpr.Name.Equals("some");
+                        foundThings |= nameExpr.Name.Equals("things");
+                        break;
+                }
+            }));
+            foundSome.Should().BeTrue();
+            foundThings.Should().BeFalse();
+        }
+
+        [TestMethod, Priority(0)]
+        public void FormattedStringConcat() {
+            var code = "f'concat of {some}' f'{other}'";
+            var parser = Parser.CreateParser(MakeReader(code), PythonLanguageVersion.V36);
+            var ast = parser.ParseFile();
+            var foundSome = false;
+            var foundOther = false;
+            ast.Walk(new MyPythonWalker(expr => {
+                switch (expr) {
+                    case NameExpression nameExpr:
+                        foundSome |= nameExpr.Name.Equals("some");
+                        foundOther |= nameExpr.Name.Equals("other");
+                        break;
+                }
+            }));
+            foundSome.Should().BeTrue();
+            foundOther.Should().BeTrue();
+        }
+
+
+        [TestMethod, Priority(0)]
+        [ExpectedException(typeof(Exception))]
+        public void CloseWithSingleBrace() {
+            var code = "f'{{ mistake}'";
+            var parser = Parser.CreateParser(MakeReader(code), PythonLanguageVersion.V36);
+            var ast = parser.ParseFile();
+        }
+
+        [TestMethod, Priority(0)]
+        public void NoEndOfEscapedBraces() {
+            var code = "f'{{ mistake'";
+            var parser = Parser.CreateParser(MakeReader(code), PythonLanguageVersion.V36);
+            var ast = parser.ParseFile();
+        }
+
+        public TextReader MakeReader(string s) => new StringReader(s);
+    }
+
+    internal class MyPythonWalker : PythonWalker {
+        private readonly Action<Expression> _t;
+        public MyPythonWalker(Action<Expression> t) {
+            _t = t;
+        }
+
+        public override bool Walk(FStringExpression node) {
+            _t(node);
+            return true;
+        }
+
+        public override bool Walk(ConstantExpression node) {
+            _t(node);
+            return true;
+        }
+
+        public override bool Walk(NameExpression node) {
+            _t(node);
+            return true;
+        }
+    }
+}
```

Comment:
I'd add some tests for longer constructs like function calls. I.e. `f"output:{func(a, b, c) + 1 + func(1, 2, 3) + C().x}" with some legal as well as not legal code fragments (i.e. try defining new class or function or import inside `{ }`